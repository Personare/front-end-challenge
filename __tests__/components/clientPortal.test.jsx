import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import 'jest-styled-components';

import ClientPortal from '../../components/clientPortal';

describe('ClientPortal', () => {
    it('Quando o ClientPortal é montado, transporta seus filhos dentro de uma div como filho direto do elemento main.', () => {
        const { asFragment } = render(
            <main>
                <div>
                    <ClientPortal>
                        <span>Tudo bom?</span>
                    </ClientPortal>
                    <span>Salve!</span>
                </div>
            </main>
        );

        expect(asFragment()).toMatchSnapshot();
    });

    it('Quando o ClientPortal transporta seus filhos, a div pai deve receber suas props.', () => {
        const { asFragment } = render(
            <main>
                <div>
                    <ClientPortal test-id="teste">
                        <span>Tudo bom?</span>
                    </ClientPortal>
                    <span>Salve!</span>
                </div>
            </main>
        );

        expect(asFragment()).toMatchSnapshot();
    });

    it('Quando o ClientPortal é montado, mas não encontra o elemento main, não renderiza nada.', () => {
        const { asFragment } = render(
            <div>
                <ClientPortal test-id="teste">
                    <span>Salve!</span>
                </ClientPortal>
            </div>
        );

        expect(asFragment()).toMatchSnapshot();
    });
});
