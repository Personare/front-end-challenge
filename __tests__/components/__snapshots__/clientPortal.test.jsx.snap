// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClientPortal Quando o ClientPortal transporta seus filhos, a div pai deve receber suas props. 1`] = `
<DocumentFragment>
  <main>
    <div>
      <span>
        Salve!
      </span>
    </div>
    <div
      test-id="teste"
    >
      <span>
        Tudo bom?
      </span>
    </div>
  </main>
</DocumentFragment>
`;

exports[`ClientPortal Quando o ClientPortal é montado, mas não encontra o elemento main, não renderiza nada. 1`] = `
<DocumentFragment>
  <div />
</DocumentFragment>
`;

exports[`ClientPortal Quando o ClientPortal é montado, transporta seus filhos dentro de uma div como filho direto do elemento main. 1`] = `
<DocumentFragment>
  <main>
    <div>
      <span>
        Salve!
      </span>
    </div>
    <div>
      <span>
        Tudo bom?
      </span>
    </div>
  </main>
</DocumentFragment>
`;
